DataManagementSystem/DB/FileCatalogComponents/DirectoryMetadata.py

```diff
@@ -57,26 +85,27 @@ def addMetadataField(self, pname, ptype, credDict):
       valueType = "VARCHAR(64)"
 
     req = "CREATE TABLE FC_Meta_%s ( DirID INTEGER NOT NULL, Value %s, PRIMARY KEY (DirID), INDEX (Value) )" \
-        % (pname, valueType)
+        % (fqPname, valueType)
```

Comment:
I think there are VOs with hyphens in their name, so the TableName now needs to be surrounded by backticks, doesn't it?

Reply:
This is not the only place which suffers from this problem. FileMetadata.py has similar statements all over the place. The problem existed even before, if a user decided to name his metadata using hyphens. I can bactick all the occurrences if you find it  a right way to go, although a really robust solution would be to use sqlalchemy, like in other Dirac modules. This would be a big change, though.

Reply:
At the moment hyphens in metadata names are not supported, which people can find a workaround. It might be harder to rename VOs for that reason.
Maybe instead of adding backticks everywhere one can replace hyphens in the getMetaNameSuffix method?